test/OAuth2AdapterFactoryTest.php

```diff
@@ -52,8 +68,9 @@ public function testInvokeWithResourceServerEmptyResponse()
             ->willReturn($this->resourceServer->reveal());
 
         $this->container
-            ->has(ResponseInterface::class)
-            ->willReturn(false);
+            ->get(ResponseInterface::class)
+            ->willReturn(function () {
+            });
```

Comment:
Also here we can use `this->responseFactory`

Reply:
No, we cannot, as it is not defined in this class.

In point of fact, this test likely needs to be re-written, as it's supposedly testing that the service can be created if the response service is missing. We should either test that an exception occurs if the service is missing (as `get()` will raise an exception), or that a `TypeError` occurs due to an invalid service. I'll make those changes shortly.

Reply:
> No, we cannot, as it is not defined in this class.

Actually it is - line 41: https://github.com/zendframework/zend-expressive-authentication-oauth2/pull/17/files/f9ce613482a2dbf64bb841c7619b3d33f39890b4#diff-d706f562446541d43749de5bbfed80b7R41